ext/src/zpages/tracez_processor.cc

```diff
@@ -6,19 +6,19 @@ namespace zpages {
 
   void TracezSpanProcessor::OnStart(opentelemetry::sdk::trace::Recordable &span) noexcept {
     std::lock_guard<std::mutex> lock(mtx_);
-    spans_.running.insert(static_cast<opentelemetry::sdk::trace::SpanData*>(&span));
+    spans_.running.insert(static_cast<opentelemetry::ext::zpages::ThreadsafeSpanData*>(&span));
   }
 
   void TracezSpanProcessor::OnEnd(std::unique_ptr<opentelemetry::sdk::trace::Recordable> &&span) noexcept {
```

Comment:
Could you change this function to remove `nullptr` from running spans? It was a case I didn't account for correctly, so running spans pointer that gets deleted don't get removed from the running container without this change:
```
void TracezSpanProcessor::OnEnd(std::unique_ptr<opentelemetry::sdk::trace::Recordable> &&span) noexcept {
    auto span_raw = static_cast<opentelemetry::ext::zpages::ThreadsafeSpanData*>(span.get());
    std::lock_guard<std::mutex> lock(mtx_);
    auto span_it = spans_.running.find(span_raw);
    if (span_it != spans_.running.end()) {
      spans_.running.erase(span_it);
      if (span != nullptr) {
        spans_.completed.push_back(
            std::unique_ptr<opentelemetry::ext::zpages::ThreadsafeSpanData>(
                static_cast<opentelemetry::ext::zpages::ThreadsafeSpanData*>(span.release())));
      }
   }
}
```

Reply:
I can do it but feel that it should be done as part of a separate PR.